src/main/java/org/gradle/playframework/plugins/PlayWebJarsPlugin.java

```diff
@@ -0,0 +1,81 @@
+    public static final String WEBJARS_EXTRACT_TASK_NAME = "extractPlayWebJars";
+
+    @Override
+    public void apply(Project project) {
+        createWebJarConfiguration(project);
+        Configuration webJarsExtractorConfiguration = createWebJarsExtractorConfiguration(project);
+        declareDefaultDependencies(project, webJarsExtractorConfiguration);
+        createDefaultWebJarsExtractTask(project, webJarsExtractorConfiguration);
+        configureWebJarsExtractTask(project);
+    }
+
+    private void createWebJarConfiguration(Project project) {
+        Configuration configuration = project.getConfigurations().create(WEBJAR_CONFIGURATION_NAME);
+        configuration.setTransitive(true);
+    }
+
+    private Configuration createWebJarsExtractorConfiguration(Project project) {
+        Configuration configuration = project.getConfigurations().create(WEBJARS_EXTRACTOR_CONFIGURATION_NAME);
+        configuration.setVisible(false);
+        configuration.setTransitive(true);
+        configuration.setDescription("The WebJars extractor library used to extract WebJars.");
+        return configuration;
+    }
+
+    private void declareDefaultDependencies(Project project, Configuration configuration) {
+        configuration.defaultDependencies(dependencies -> {
+            dependencies.add(project.getDependencies().create(WebJarsExtractor.getDependencyNotation()));
+        });
+    }
+
+    private void createDefaultWebJarsExtractTask(Project project, Configuration configuration) {
+        project.getTasks().register(WEBJARS_EXTRACT_TASK_NAME, WebJarsExtract.class, task -> {
+            task.setDescription("Extracts WebJars.");
+            task.getOutputDirectory().set(project.getLayout().getBuildDirectory().dir(GENERATED_SOURCE_ROOT_DIR_PATH + "/webJars"));
+            task.getWebJarsClasspath().setFrom(project.getConfigurations().getByName(WEBJAR_CONFIGURATION_NAME));
+            task.getWebJarsExtractorClasspath().setFrom(configuration);
+        });
+    }
+
+    private void configureWebJarsExtractTask(Project project) {
+        TaskProvider<WebJarsExtract> webJarsExtractTaskProvider = project.getTasks().named(WEBJARS_EXTRACT_TASK_NAME, WebJarsExtract.class);
+
+        project.getTasks().named(ASSEMBLE_TASK_NAME, task -> {
+            task.dependsOn(webJarsExtractTaskProvider);
+        });
+        project.getTasks().named(ASSETS_JAR_TASK_NAME, Jar.class, task -> {
+            task.dependsOn(webJarsExtractTaskProvider);
+            task.from(webJarsExtractTaskProvider.get().getOutputDirectory(), copySpec -> copySpec.into("public"));
+        });
+        project.getTasks().named(PLAY_RUN_TASK_NAME, PlayRun.class, task -> {
+            task.getAssetsDirs().from(webJarsExtractTaskProvider.get().getOutputDirectory());
+        });
+        project.getTasks().matching(task -> task.getName().equals(LESS_COMPILE_TASK_NAME)).forEach(task -> {
```

Comment:
This forces the realization of all tasks. Can you use a type constraint instead of a name constraint?

Reply:
Ah, just realized this. Changed this to:

```
Optional.ofNullable(project.getTasks().findByName(LESS_COMPILE_TASK_NAME)).ifPresent(task -> {
```

Can't find any method which accepts type constraint but does not fail if the task is not found :(

Reply:
Asking internally about this. I think that `findByName` is still forcing some task configuration & realization?

Reply:
To be honest, I'm not really sure about this. The thing is I cannot use `.named()` because it will throw if the less plugin is not applied. Most of the methods I found is doing the similar thing (throwing when task is not found). 

Could you advise the correct method? I could not find anything by browsing the Task javadoc / documentation :(

Reply:
After asking internally, the answer I got was `tasks.matching { ... }.configureEach { ... }` This may be what you already had (sorry). But supposedly, this is what you want to use here.

Reply:
I see. No worries, I added it back!